packages/react/src/generators/component/component.ts

```diff
@@ -106,10 +106,19 @@ function addExportsToBarrel(host: Tree, options: NormalizedSchema) {
     workspace.get(options.projectName).projectType === 'application';
 
   if (options.export && !isApp) {
-    const indexFilePath = joinPathFragments(
+    let indexFilePath = joinPathFragments(
       options.projectSourceRoot,
       options.js ? 'index.js' : 'index.ts'
     );
+
+    if (!host.exists(indexFilePath)) {
+      indexFilePath = joinPathFragments(
+        options.projectSourceRoot,
+        'src',
```

Comment:
Is this correct? It could produce a duplicated `src` segment, isn't it?

Reply:
If `sourceRoot` is set as `<project>/src` then the above clause would work. This is handling cases where we default `projectSourceRoot` to `projectRoot`, which is always the case in inferred projects. In this case we still want to pick up`<projectRoot>/src/index.ts`, which is why this logic exists.

I updated `normalizeOptions` to keep both `projectRoot` and `projectSourceRoot` so the logic here is more clear.